File: backend/app/schemas/video.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


VideoStatus = Literal[
    "uploading",
    "processing",
    "ready",
    "failed",
    "deleted",
]


class VideoCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)

    description: str | None = None

    storage_path: str

    mime_type: str

    file_size_bytes: int = Field(ge=0)


class VideoResponse(BaseModel):
    id: str

    name: str

    description: str | None = None

    storage_path: str

    public_url: str | None = None

    thumbnail_url: str | None = None

    mime_type: str

    file_size_bytes: int

    duration_seconds: float

    status: VideoStatus

    created_at: datetime

    updated_at: datetime

File: frontend/types/video.ts
export type VideoStatus =
  | "uploading"
  | "processing"
  | "ready"
  | "failed"
  | "deleted";

export interface Video {
  id: string;

  name: string;

  description?: string | null;

  storagePath: string;

  publicUrl?: string | null;

  thumbnailUrl?: string | null;

  mimeType: string;

  fileSizeBytes: number;

  fileSizeMb: number;

  durationSeconds: number;

  width?: number | null;

  height?: number | null;

  status: VideoStatus;

  createdAt: string;

  updatedAt: string;
}

export interface CreateVideoInput {
  name: string;

  description?: string;

  storagePath: string;

  mimeType: string;

  fileSizeBytes: number;
}

export interface VideoUploadProgress {
  videoId: string;

  progress: number;

  status: VideoStatus;

  error?: string | null;
}
